### ISSUE TITLE
bug in error reporting

### ISSUE THREAD

**Issue by jacekm-git** [orangeduck/mpc#41]
Hi,
instead of detailed error message, generic "Unknown Error"  is displayed in some cases. Bug appeared in commit 
227dd44.

To be specific:
i'm working on code from chapter 6 build your own lisp.

example:
````
lispy>xxx
<stdinput>: error: Unknown Error
````
this works:
````
lispy> + 1 x
<stdinput>:1:5: error: expected '-', one or more of one of '0123456789', '(' or end of input at 'x'
````

**Comment by orangeduck** [comment]
Thanks for the report I'll look into it!

**Comment by orangeduck** [comment]
I was unable to reproduce this with the current master of `mpc`. Perhaps you can try with the master and also the `parsing.c` source code from the book - there might be a problem in your implementation or an older version of mpc you are using.

**Comment by jacekm-git** [comment]
How to reproduce:
```sh                                                                                                     
git clone https://github.com/orangeduck/mpc.git
git clone https://github.com/orangeduck/BuildYourOwnLisp.git
cp mpc/mpc.* BuildYourOwnLisp/src/
cd BuildYourOwnLisp/src  
```
Optional step. I need this because there is no history.h on my system.
```sh
sed -i s/"#include <editline\/history.h>"//g *.c
```

```sh
make                                      
 ./parsing
Lispy Version 0.0.0.0.2
Press Ctrl+c to Exit

 lispy> 1
 <stdin>: error: Unknown Error
``` 

My specs
```sh
uname -a
Linux laptop 4.2.5-1-ARCH #1 SMP PREEMPT Tue Oct 27 08:13:28 CET 2015 x86_64 GNU/Linux

gcc --version
gcc (GCC) 5.3.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by orangeduck** [comment]
Thanks for the detailed steps. I managed to reproduce the error. Sorry my bad - my local version had some differences.

This should be fixed in 3ead545ed82932074c51a5567d6598b88a2d0178.

Hopefully you can verify.

Thanks,

- Dan

**Comment by jacekm-git** [comment]
Yes, I can confirm, everything works well. Thanks.
